### PR TITLE
Added valueOf definitions for String and Number

### LINKED ISSUE #1849: Lib.d.ts should define overrides of `valueOf` for String, Number and Boolean
Hi,

**TS**: 1.4

The compiler issues error "Type 'Object' is not assignable to type 'string'" for the following:

```TypeScript
var f: string = new String('foo').valueOf();
```

This is because `valueOf` resolves to the definition on `Object` rather than `String` - which is currently missing.

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/valueOf):

> By default, the valueOf method is inherited by every object descended from Object. Every built-in core object overrides this method to return an appropriate value.

Is there any reason why `valueOf` is missing in the definitions for `String`, `Number` and `Boolean`? (The `Date` type seems to have one defined.)

Related issues:
#1387

**Edit**:

I understand that we want to suppress the display of the standard methods:

```TypeScript
    toString(): string;
    toLocaleString(): string;
    valueOf(): Object;
    hasOwnProperty(v: string): boolean;
    isPrototypeOf(v: Object): boolean;
    propertyIsEnumerable(v: string): boolean;
```

But unlike the other methods the return value of `valueOf` is specific to the type.

### PR DESCRIPTION
**Pull request by DickvdBrink**
Fixes #1849  I guess :)

edit: Oops, I did some manual testing and it worked fine (and because it was such a minor change)  I didn't run Jake runtests. Sorry. Will take a look

### CODE REVIEW

**Comment by mhegazy** [comment]
Just some error baselines changed. you will need to accept them. :+1:

**Comment by DickvdBrink** [comment]
@mhegazy, some completion stuff changed to so I need to change some test. Will do that asap (hopefully I can to it today :))

File: tests/cases/fourslash/completionEntryForUnionProperty2.ts [hunk 1]
```diff
@@ -16,4 +16,5 @@
 
 goTo.marker();
 verify.memberListContains("toString", "(method) toString(): string");
-verify.memberListCount(1);
\ No newline at end of file
+verify.memberListContains("valueOf", "(method) valueOf(): string | number");
```
Comment (DickvdBrink) [inline tests/cases/fourslash/completionEntryForUnionProperty2.ts]: Did you guys expected this?

Comment (DickvdBrink) [inline tests/cases/fourslash/completionEntryForUnionProperty2.ts]: Did you guys expect this?

File: tests/cases/fourslash/completionEntryForUnionProperty2.ts [hunk 2]
```diff
@@ -16,4 +16,5 @@
 
 goTo.marker();
 verify.memberListContains("toString", "(method) toString(): string");
-verify.memberListCount(1);
\ No newline at end of file
+verify.memberListContains("valueOf", "(method) valueOf(): string | number");
+verify.memberListCount(2);
```
Comment (DickvdBrink) [inline tests/cases/fourslash/completionEntryForUnionProperty2.ts]: Was this expected?

**Comment by mhegazy** [comment]
thanks!

**Comment by NoelAbrahams** [comment]
Don't we need one for `Boolean`?

**Comment by DickvdBrink** [comment]
@NoelAbrahams, that one is already there :)

**Pull request merged**